cmd/kinflate/README.md

```diff
@@ -25,14 +21,79 @@ control commits.  It encourages forking a configuration,
 customizing it, and occasionally rebasing to capture
 upgrades from the original configuration.
 
+
 ## Installation
 
+`kinflate` is written in Go and can be installed using `go get`:
+
 <!-- @installKinflate @test -->
+```shell
+go get -u k8s.io/kubectl/cmd/kinflate
 ```
-go get k8s.io/kubectl/cmd/kinflate
+
+After running the above command, `kinflate` should get installed in your `GOPATH/bin` directory.
+
+## Usage
+
+This section explains sub-commands of `kinflate` using examples. 
+
+### init 
+`kinflate init` initializes the Kubernetes Application by creating an
+application manifest file named `Kube-manifest.yaml` in current directory. This
+should be run in the same directory where application config resides.
+
+```shell
+kinflate init
 ```
 
+### set nameprefix
+
+`kinflate set nameprefix` updates the nameprefix field in the [manifest].
+`kinflate inflate` uses the prefix to generate prefixed names for Kubernetes resources.
+
+#### examples
+```shell
+# set nameprefix to 'staging'
+kinflate set nameprefix staging
+
+# set nameprefix to 'john`
+kinflate set nameprefix john
+```
+
+### add 
+`kinflate add` adds a resource, configmap or secret to the [manifest]. 
+
+#### examples
+```shell
+# adds a resource to the manifest
+kinflate add resource <path/to/resource-file-or-dir>
+
+# adds a configmap from a literal or file-source to the manifest
+kinflate add configmap my-configmap --from-file=my-key=file/path --from-literal=my-literal=12345
+
+# Adds a generic secret to the Manifest (with a specified key)
+kinflate add secret generic my-secret --from-file=my-key=file/path --from-literal=my-literal=12345
+
+# Adds a TLS secret to the Manifest (with a specified key)
+kinflate add secret tls my-tls-secret --cert=cert/path.cert --key=key/path.key        
+```
+
+### inflate
+`kinflate inflate` generates application configuration by applying
+transformations (e.g. name-prefix, label, annotations) specified in the [manifest].
+
+#### examples
+```shell
+# generate application configuration and output to stdout
+kinflate inflate -f <path/to/application/dir>
+
+# generate application configuration and apply it to a Kubernetes cluster
+kinflate inflate -f <path/to/application/dir> | kubectl apply -f
+```
+
+
 ## Demos and tutorial
 
+ * [Getting Started](getting_started.md)
  * [Short demo](shortDemo.md)
```

Comment:
"short demo", "long demo" titles are not useful. The type of example titles I'd like to see:

- Using kinflate to inject environment variable
- Using kinflate to modify annotation
- Using kinflate to declare ports
- Using kinflate to [...]

Reply:
Great point. Thats more aligned with what users would be looking for. Added a note, will be adding in next set of changes.